src/qiboml/operations/differentiation.py

```diff
@@ -1,17 +1,21 @@
+from typing import Optional, Union
+
 import numpy as np
-from qibo.backends import construct_backend
+import qibo
+import qibo.backends
 from qibo.config import raise_error
 from qibo.hamiltonians.abstract import AbstractHamiltonian
+from qibojit.backends import NumbaBackend
 
 
-def parameter_shift(
-    hamiltonian,
-    circuit,
-    parameter_index,
-    initial_state=None,
-    scale_factor=1,
-    nshots=None,
-    backend="qibojit",
+def one_parameter_shift(
```

Comment:
I think this is partially true. 

1. it can be useful (in some situations) to compute the gradient w.r.t. a specific parameter only. 
2. ofc one can extract one gradient from the gradients list, but, if this is basically _for free_ using symbolical differentiation, in the case of using other rules (e.g. the parameter shift), the computational cost is drastically different.

Reply:
Yes, but as I wrote, the way to go in that case would be to set a single trainable parameter.

Maybe it's not the most comfortable approach (though we can make a helper function that takes a circuit, the parameter specification, and return another circuit, being the exact same but with that single parameter marked as *trainable*), but there is no computational overhead, giving you access to the exact same computation.
The great advantage is to have a single uniform interface, and this will be highly beneficial.

Reply:
Well well well, I got the point now. I agree, thanks for the suggestion!